www/js/lib/zimArchive.js

```diff
@@ -135,7 +137,27 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
     ZIMArchive.prototype.isReady = function() {
         return this._file !== null;
     };
-    
+
+    /**
+     * Detects whether the supplied archive is a Zimit-style archive or an OpenZIM archive and
+     * sets a file.type property accordingly; also returns the detected type. Extends ZIMFile.
+     * @returns {String} Either 'zimit' for a Zimit archive, or 'open' for an OpenZIM archive
+     */
+     ZIMArchive.prototype.setZimType = function () {
+        var fileType = null;
+        if (this.isReady()) {
+            fileType = 'open';
+            this._file.mimeTypes.forEach(function (v) {
+                if (/warc-headers/i.test(v)) fileType = 'zimit';
```

Comment:
It's very minor but check if you want to match `warc-headers` anywhere in the mimeType string, or exactly this string. In this case it would be safer to use:
```suggestion
                if (/^warc-headers$/i.test(v)) fileType = 'zimit';
```

Reply:
I think it is best to keep this broad because any appearance of `warc-headers` in a MIME type indicates we have a ZIM that was created with warc2zim. The full string is currently `application/warc-headers`, but there's no particular reason why `text/warc-headers` (for example) would be invalid, since the headers are in text format. But we might in future have `application/warc-headers-encoded` (there's no guarantee of stability, as there is no published spec), so making the test too specific could limit the ability to detect a Zimit ZIM in this case.

Reply:
Sure, no problem